src/sonic-yang-mgmt/sonic_yang_ext.py

```diff
@@ -446,23 +491,60 @@ def revXlateList(self, model, yang, config, table):
+    Rev xlate a list inside a yang container
+    """
+    def revXlateListInContainer(self, model, yang, config, table):
+        modelList = model
+        # Pass matching list from Yang Json if exist
+        if yang.get(modelList['@name']):
+            self.sysLog(msg="revXlateListInContainer {}".format(modelList['@name']))
+            self.revXlateList(modelList, yang[modelList['@name']], config, table)
+        return
+
+    """
+    Rev xlate a container inside a yang container
+    """
+    def revXlateContainerInContainer(self, model, yang, config, table):
+        modelContainer = model
+        # Pass matching list from Yang Json if exist
+        if yang.get(modelContainer['@name']):
+            config[modelContainer['@name']] = dict()
+            self.sysLog(msg="revXlateContainerInContainer {}".format(modelContainer['@name']))
+            self.revXlateContainer(modelContainer, yang[modelContainer['@name']], \
+                config[modelContainer['@name']], table)
+        return
+
     """
     Rev xlate from yang container to table in config DB
     """
     def revXlateContainer(self, model, yang, config, table):
 
-        # Note: right now containers has only LISTs.
         # IF container has only one list
-        if isinstance(model['list'], dict):
-            modelList = model['list']
-            # Pass matching list from Yang Json
-            self.sysLog(msg="revXlateContainer {}".format(modelList['@name']))
-            self.revXlateList(modelList, yang[modelList['@name']], config, table)
-
-        elif isinstance(model['list'], list):
-            for modelList in model['list']:
-                self.sysLog(msg="revXlateContainer {}".format(modelList['@name']))
-                self.revXlateList(modelList, yang[modelList['@name']], config, table)
+        clist = model.get('list')
+        if isinstance(clist, dict):
+            self.revXlateListInContainer(clist, yang, config, table)
+        # IF container has lists
+        elif isinstance(clist, list):
+            for modelList in clist:
+                self.revXlateListInContainer(modelList, yang, config, table)
+
+        ccontainer = model.get('container')
+        # IF container has only one inner container
+        if isinstance(ccontainer, dict):
+            self.revXlateContainerInContainer(ccontainer, yang, config, table)
+        # IF container has only many inner container
+        elif isinstance(ccontainer, list):
+            for modelContainer in ccontainer:
+                self.revXlateContainerInContainer(modelContainer, yang, config, table)
+
+        ## Handle other leaves in container,
+        leafDict = self.createLeafDict(model)
+        for vKey in yang:
+            #vkey must be a leaf\leaf-list\choice in container
+            if leafDict.get(vKey):
+                self.sysLog(syslog.LOG_DEBUG, "revXlateContainer vkey {}".format(vKey))
+                config[vKey] = self.revFindYangTypedValue(vKey, yang[vKey], leafDict)
 
```

Comment:
Supports one level of containers/lists in container?   Any need to support more levels of containers/lists in containers?

Reply:
Multiple lists case exists of INTERFACE and VLAN_INTERFACE Table after the VRF feature. 

Multiple containers case does not exist as of now, but I thought to support it, as in future, the conversion will fail if someone will write yang models with >=2 containers in another container.